cli/src/commands/git/init.rs

```diff
@@ -172,18 +173,23 @@ fn do_init(
             maybe_add_gitignore(&workspace_command)?;
             workspace_command.maybe_snapshot(ui)?;
             maybe_set_repository_level_trunk_alias(ui, &workspace_command)?;
-            if !workspace_command.working_copy_shared_with_git() {
-                let mut tx = workspace_command.start_transaction();
+            let working_copy_shared_with_git = workspace_command.working_copy_shared_with_git();
+            let mut tx = workspace_command.start_transaction();
+            if !working_copy_shared_with_git {
                 jj_lib::git::import_head(tx.repo_mut())?;
                 if let Some(git_head_id) = tx.repo().view().git_head().as_normal().cloned() {
                     let git_head_commit = tx.repo().store().get_commit(&git_head_id)?;
                     tx.check_out(&git_head_commit)?;
                 }
-                if tx.repo().has_changes() {
-                    tx.finish(ui, "import git head")?;
-                }
             }
-            print_trackable_remote_bookmarks(ui, workspace_command.repo().view())?;
+            if settings.get_bool("git.init-track-local-bookmarks")? {
+                track_local_bookmarks(ui, tx.repo_mut())?;
+            } else {
+                print_trackable_remote_bookmarks(ui, tx.repo().view())?;
```

Comment:
Seems like this function could be moved into this file. It seems better to have `track_local_bookmarks()` and `print_trackable_remote_bookmarks()` near each other, don't you think?